plugin/main/src/kotlinx/benchmark/gradle/NativeMultiplatformTasks.kt

```diff
@@ -74,14 +74,8 @@ private fun Project.createNativeBenchmarkCompileTask(target: NativeBenchmarkTarg
         sourceSet.resources.setSrcDirs(files())
         sourceSet.kotlin.setSrcDirs(files("$benchmarkBuildDir/sources"))
 
-        sourceSet.dependencies {
-            implementation(compilation.output.allOutputs)
```

Comment:
@sellmair could you please clarify why it solves the issue?

I'm looking into `KotlinNativeCompilationAssociator`, and its implementation looks almost the same: https://github.com/JetBrains/kotlin/blob/ad8cea5f0c154598a45bc8f720c07d8de9fe19f1/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/plugin/mpp/compilationImpl/KotlinCompilationAssociator.kt#L69.

Reply:
Of course! You see the main difference is that kotlinx.benchmark used the `implementation` configuration to attach the dependencies. This will be resolved by the IDE for dependencies. However, the `compileDependencyFiles` or `compileDependencyConfiguration` is at the very end of the dependency-resolution pipeline and will not cause the IDE resolve this.

Reply:
So in short: `implementation` is "for users declaring dependencies"
'compileDependencyFiles/compileDependencyConfiguration' is the actual 'lower level' container for the compile classpath

Reply:
Thanks!